src/lib/updateContractInfo.ts

```diff
@@ -0,0 +1,191 @@
+import 'dotenv/config'
+import 'module-alias/register'
+import 'source-map-support/register'
+
+import { BlockWithTransactions } from '@ethersproject/abstract-provider'
+import Bluebird from 'bluebird'
+import { differenceInDays, isAfter } from 'date-fns'
+import { Contract, BigNumber as BN } from 'ethers'
+import { Op } from 'sequelize'
+import knownContracts from '../common/knownContracts'
+import { CONTRACT_INFO_MAX_AGE_IN_DAYS } from './constants'
+import getContractSpec from './getContractSpec'
+import { getEthClient } from './getEthClient'
+import Logger from './logger'
+import {
+  IContractInfoJobDetailsByTokenAddress,
+  IContractInfoJobDetailsByBlock
+} from './queueContractInfoJobs'
+import safeCall from './safeCall'
+import stats from './stats'
+import { ContractInfo } from '@models/index'
+import { ProcessingGoal } from '@types'
+
+const abi = [
```

Comment:
Maybe this can be abstracted elsewhere and reused since `updateTokenInfo` also makes use of these function signatures

Reply:
maybe... they are different readable functions but yeah. can check.